builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -355,9 +371,12 @@ func resourceVSphereVirtualMachineRead(d *schema.ResourceData, meta interface{})
 
 	vm, err := finder.VirtualMachine(context.TODO(), d.Get("name").(string))
 	if err != nil {
-		log.Printf("[ERROR] Virtual machine not found: %s", d.Get("name").(string))
 		d.SetId("")
-		return nil
+		return fmt.Errorf("[ERROR] Virtual machine not found: %s", d.Get("name").(string))
```

Comment:
Was there a reason you had to change this to make folder VMs work? It's important for `Read` not to return an error when the VM is not found, otherwise if a VM is deleted for any other reason, Terraform will be unable to make progress in any future `plan` or `apply` operations, since they'll return this error.